docs/adr/0009-naming-conventions.md

```diff
@@ -0,0 +1,158 @@
+# Naming conventions
+
+Date: 2020-09-24
+
+## Status
+
+2020-12-02 Proposed
+
+## Component files
+
+### Principles
+
+- Keep it simple, reduce nesting
+- A `src/pages` folder will hold 'root' components and reflect the navigation structure (menu, routes) of the app
+- Components needed only by one 'root' component go into that root component's folder, no extra nesting
+- Components needed by more than one 'root' component go into `src/components`
+- Component folders, files, and auxiliary files, camel cased, starting with upper case
+- Component names and file names will be unique
+- Generic/shared components, hooks, interfaces, services, in their respective folders in `src`.
+
+### Example tree
+
+```
+src
+  pages
+    login
+      Login.tsx
+    manage
+      default-texts
+        DefaultTexts.tsx
+        TextsForm.tsx
+        SelectForm.tsx
+      incidents
+        incident
+          split
+            IncidentSplit.tsx
+            SplitForm.tsx
+            SplitFormIncident.tsx
+          Incident.tsx
+        Incidents.tsx
+        IncidentsList.tsx
+        IncidentsMap.tsx
+        Filter.tsx
+        FilterTagList.tsx
+    report
+      Incident.tsx
+      IncidentForm.tsx
+      IncidentNavigation.tsx
+      IncidentPreview.tsx
+      IncidentWizard.tsx
+    kto
+      Kto.tsx
+      KtoForm.tsx
+    settings
+      users
+        user
+          User.tsx
+        Users.tsx
+      roles
+        role
+          Role.tsx
+        Roles.tsx
+      departments
+        department
+          Department.tsx
+          CategoryGroups.tsx
+          CategoryLists.tsx
+        Departments.tsx
+      categories
+        category
+          Category.tsx
+          CategoryForm.tsx
+        Categories.tsx
+    App.tsx
+    App.styles.ts
+    App.test.ts
+  components
+    RadioInput
+      RadioInput.tsx
+      RadioInput.styles.ts
+      RadioInput.test.ts
+    SelectInput
+      SelectInput.tsx
+      SelectInput.styles.ts
+      SelectInput.test.ts
+    List
+      List.tsx
+      List.styles.ts
+      List.test.ts
+  hooks
+    useFetch.ts
+    useFetch.test.ts
+  interfaces
+  services
+```
+
+This gives a clear hierarchy without much nesting.
+
+## Styling
```

Comment:
I thought about this and I don't agree with imposing this construction for many reasons, few of these would be: 
- loosing the capability (or making difficult) implementing small styles inside a component. I have further searched the internet and I don't see a trend in this direction.
- Is almost impossible to convert the whole codebase to this format in a acceptable period of time
- Requires for almost each component an extra file.